google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -162,16 +161,13 @@ protected void hideWarning() {
       public void run() {
         cloudSdkDirectoryField.getTextField().setForeground(JBColor.black);
         warningIcon.setVisible(false);
-        warningIcon.setVisible(false);
         warningMessage.setVisible(false);
-        cloudSdkPanel.updateUI();
       }
     });
   }
 
   private void invokePanelValidationUpdate(Runnable runnable) {
-    ApplicationManager.getApplication().invokeLater(runnable,
-        ModalityState.stateForComponent(cloudSdkPanel));
+    ApplicationManager.getApplication().invokeLater(runnable, ModalityState.current());
```

Comment:
So, even though `ModalityState`'s documentation discourages the use of `any()` ("unless absolutely needed" they say), I found it was the only modality state where the message gets correctly drawn when the settings window is opened.